src/timeline/Timeline.js

```diff
@@ -0,0 +1,216 @@
+// @flow
+import {
+  View,
+  Text,
+  ScrollView,
+  TouchableOpacity,
+  Dimensions,
+} from 'react-native';
+import populateEvents from './Packer';
+import React from 'react';
+import moment from 'moment';
+import _ from 'lodash';
+import styleConstructor from './style';
+
+const LEFT_MARGIN = 60 - 1;
+// const RIGHT_MARGIN = 10
+// const CALENDER_HEIGHT = 2400;
+// const EVENT_TITLE_HEIGHT = 15
+const TEXT_LINE_HEIGHT = 17;
+// const MIN_EVENT_TITLE_WIDTH = 20
+// const EVENT_PADDING_LEFT = 4
+
+function range(from, to) {
+  return Array.from(Array(to), (_, i) => from + i);
+}
+
+let {width: dimensionWidth} = Dimensions.get('window');
+
+export default class Timeline extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const {start = 0, end = 24} = this.props;
+    this.styles = styleConstructor(props.styles, (end - start) * 100);
+    this.calendarHeight = (end - start) * 100;
+    const width = dimensionWidth - LEFT_MARGIN;
+    const packedEvents = populateEvents(props.events, width, start);
+    let initPosition =
+      _.min(_.map(packedEvents, 'top')) - this.calendarHeight / (end - start);
+    initPosition = initPosition < 0 ? 0 : initPosition;
+    this.state = {
+      _scrollY: initPosition,
+      packedEvents,
+    };
+  }
+
+  componentWillReceiveProps({events, start = 0}) {
+    const width = dimensionWidth - LEFT_MARGIN;
+    this.setState({
+      packedEvents: populateEvents(events, width, start),
+    });
+  }
+
+  componentDidMount() {
+    this.props.scrollToFirst && this.scrollToFirst();
+  }
+
+  scrollToFirst() {
+    setTimeout(() => {
+      if (this.state && this.state._scrollY && this._scrollView) {
+        this._scrollView.scrollTo({
+          x: 0,
+          y: this.state._scrollY,
+          animated: true,
+        });
+      }
+    }, 1);
+  }
+
+  // _renderRedLine() {
+  //   const offset = 100;
+  //   const {start = 0} = this.props;
+  //   const timeNowHour = moment().hour();
+  //   const timeNowMin = moment().minutes();
+  //   return (
+  //     <View
+  //       key={'timeNow'}
+  //       style={[
+  //         this.styles.lineNow,
+  //         {
+  //           top: offset * (timeNowHour - start) + (offset * timeNowMin) / 60,
+  //           width: dimensionWidth - 20,
+  //         },
+  //       ]}
+  //     />
+  //   );
+  // }
+
+  _renderLines() {
+    const {format24h, start = 0, end = 24} = this.props;
+    const offset = this.calendarHeight / (end - start);
+
+    return range(start, end + 1).map((i, index) => {
```

Comment:
Consider displaying the time as `i:00` when not in a format24h, as just `i` is not very time-like display

Reply:
@Inbal-Tish Done!